test/test_custom_ops.py

```diff
@@ -721,6 +722,40 @@ def _generate_examples(self, typ):
             f"testrunner cannot generate instanstance of type {typ}"
         )
 
+    def test_mangle_demangle(self):
+        examples = [
+            "__main__",
+            "__main__.foo",
+            "foo.bar.baz",
+            "X.Z.XZ.ZX.ZZ",
+            "Z0ZZ1ZZZ2",
+            "torch.testing._internal.custom_op_db.fn0",
+            "torch.testing._internal.custom_op_db.get_fn1.<locals>.fn1",
+            "torch.testing._internal.custom_op_db.get_fn2.<locals>.<lambda>@0xDEADBEEF",
+        ]
+
+        for value in examples:
+            mangled = utils.mangle(value)
+            self.assertTrue(mangled.isidentifier())
+            demangled = utils.demangle(mangled)
+            self.assertEqual(demangled, value)
+
+    def test_unique_name(self):
```

Comment:
I would also test this with:
- method
- classmethod
- staticmethod
- builtin
- builtin method